platforms/macos/src/subclass.rs

```diff
@@ -0,0 +1,177 @@
+    let key = ViewKey(this as *const _);
+    let instances = INSTANCES.lock();
+    let instance = instances.get(&key).unwrap();
+    (*instance.0).prev_class.as_ref().unwrap().superclass()
+}
+
+unsafe extern "C" fn children(this: &NSView, _cmd: Sel) -> *mut NSArray<NSObject> {
+    let key = ViewKey(this as *const _);
+    let instances = INSTANCES.lock();
+    let instance = instances.get(&key).unwrap();
+    (*instance.0).adapter.view_children()
+}
+
+unsafe extern "C" fn focus(this: &NSView, _cmd: Sel) -> *mut NSObject {
+    let key = ViewKey(this as *const _);
+    let instances = INSTANCES.lock();
+    let instance = instances.get(&key).unwrap();
+    (*instance.0).adapter.focus()
+}
+
+unsafe extern "C" fn hit_test(this: &NSView, _cmd: Sel, point: NSPoint) -> *mut NSObject {
+    let key = ViewKey(this as *const _);
+    let instances = INSTANCES.lock();
+    let instance = instances.get(&key).unwrap();
+    (*instance.0).adapter.hit_test(point)
+}
+
+impl Instance {
+    fn new(view: Id<NSView, Shared>, adapter: Adapter) -> Box<Self> {
+        Box::new(Self {
+            view,
+            adapter,
+            prev_class: None,
+        })
+    }
+
+    fn install(&mut self) {
+        let view_ptr = Id::as_ptr(&self.view);
+        let key = ViewKey(view_ptr);
+        INSTANCES.lock().insert(key, InstancePtr(self as *const _));
+        // Cast to a pointer and back to force the lifetime to 'static
+        // SAFETY: We know the class will live as long as the instance,
+        // and we own a reference to the instance.
+        let superclass = unsafe { &*(self.view.class() as *const Class) };
+        self.prev_class = Some(superclass);
+        let mut subclasses = SUBCLASSES.lock();
+        let entry = subclasses.entry(superclass);
+        let subclass = entry.or_insert_with(|| {
+            let name = format!("AccessKitSubclassOf{}", superclass.name());
+            let mut builder = ClassBuilder::new(&name, superclass).unwrap();
+            unsafe {
+                builder.add_method(
+                    sel!(superclass),
+                    get_superclass as unsafe extern "C" fn(_, _) -> _,
+                );
+                builder.add_method(
+                    sel!(accessibilityChildren),
+                    children as unsafe extern "C" fn(_, _) -> _,
+                );
+                builder.add_method(
+                    sel!(accessibilityFocusedUIElement),
+                    focus as unsafe extern "C" fn(_, _) -> _,
+                );
+                builder.add_method(
+                    sel!(accessibilityHitTest:),
+                    hit_test as unsafe extern "C" fn(_, _, _) -> _,
+                );
+            }
+            builder.register()
+        });
+        unsafe { object_setClass(view_ptr as *mut _, (*subclass as *const Class).cast()) };
+    }
+
+    fn uninstall(&self) {
+        let view_ptr = Id::as_ptr(&self.view);
```

Comment:
Just FYI: Taking an immutable reference `&self.view` and mutating it like this is only possible because `NSView` effectively contains `ffi::objc_object`, which is `UnsafeCell` internally.